pinot-segment-local/src/main/java/org/apache/pinot/segment/local/indexsegment/mutable/MutableSegmentImpl.java

```diff
@@ -966,16 +966,21 @@ public Set<String> getPhysicalColumnNames() {
   public DataSource getDataSource(String column) {
     FieldSpec fieldSpec = _schema.getFieldSpecFor(column);
     if (fieldSpec == null || fieldSpec.isVirtualColumn()) {
+      VirtualColumnContext virtualColumnContext = null;
       // Column is either added during ingestion, or was initiated with a virtual column provider
       if (fieldSpec == null) {
         // If the column was added during ingestion, we will construct the column provider based on its fieldSpec to
         // provide values
         fieldSpec = _newlyAddedColumnsFieldMap.get(column);
         Preconditions.checkNotNull(fieldSpec,
             "FieldSpec for " + column + " should not be null. " + "Potentially invalid column name specified.");
+        // newly added column shouldn't have any doc count?
+        virtualColumnContext = new VirtualColumnContext(fieldSpec, 0);
```

Comment:
Why do we change this? I think the old code is correct

Reply:
newly added column should have same doc count right? just cardinality will always be 1 due to only default null value will be select out?

Reply:
ah I see. I thought the doc count here represents number of docs with non-null values for this column. 
I wasn't sure about this change. Thanks for clarifying. Let me revert to old code and verify that the fix still works.

Reply:
yep. old code works fine 👍